src/components/pages/deployments/deploymentDetails/deploymentDetailsGrid/deploymentDetailsGrid.js

```diff
@@ -19,7 +29,43 @@ export class DeploymentDetailsGrid extends Component {
     ];
   }
 
-  onGridReady = gridReadyEvent => this.deployedDevicesGridApi = gridReadyEvent.api;
+  onGridReady = gridReadyEvent => {
+    this.deployedDevicesGridApi = gridReadyEvent.api;
+    // Call the onReady props if it exists
+    if (isFunc(this.props.onGridReady)) {
+      this.props.onGridReady(gridReadyEvent);
+    }
+  };
+
+  onSoftSelectChange = (deviceRowId, rowEvent) => {
+    const { onSoftSelectChange } = this.props;
+    const rowData = (this.deployedDevicesGridApi.getDisplayedRowAtIndex(deviceRowId) || {}).data;
+    if (rowData && rowData.device) {
+      this.setState({
+        openFlyoutName: 'deviceDetails',
+        selectedDevice: rowData.device
+      });
+    } else {
+      this.closeFlyout();
+    }
+    if (isFunc(onSoftSelectChange)) {
+      onSoftSelectChange(rowData, rowEvent);
+    }
+  }
+
+  closeFlyout = () => this.setState(closedFlyoutState);
+
+  getSoftSelectId = ({ id } = {}) => id;
```

Comment:
`id` shouldn't be an object. If was copy-pasted from somewhere that should be changed too please.
```javascript
({ id } = '')
```

Reply:
fixed everywhere